content/3_Silver/Intro_Sorted_Sets.mdx

```diff
@@ -196,7 +224,75 @@ System.out.println(set.higher(23));  // ERROR, no such element exists
 ```
 
 </JavaSection>
+<PySection>
+
+Python does not have a sorted set, so see the C++ or Java if you want an
+implementation from the standard library.
+
+Creating a sorted set from scratch is time consuming, and not reccomended to do
+so in contests, but if you are curious regarding the implementation, you can
+find an AVL tree implementation of a sorted set below.
+
+<Resources>
+	<Resource source="DSA Python" url="https://nibmehub.com/opac-service/pdf/read/Data%20Structures%20and%20Algorithms%20in%20Python.pdf#page=503" title="AVL Trees">
+		definition and implementation of AVL Trees in Python
+	</Resource>
+</Resources>
+
+<Warning>
+
+You are not expected to know how to create an AVL tree in USACO Silver, nor is
+it reccomended to do so for a representation of a sorted set, since other
+languages have sorted sets built in.
+
+</Warning>
+
+Since some online judges include additional libraries, an implementation of
+sorted sets from the `sortedcontainers` library is shown below. All operations
+shown below are $\mathcal{O}(\log N)$ time, except for its
+$\mathcal{O}(N \log N)$ initialization.
+
+```py
+from sortedcontainers import SortedSet
+
+sorted_set = SortedSet([5, 1, 3, 2])
+print(sorted_set)  # SortedSet([1, 2, 3, 4, 7])
+
+# Add elements
+sorted_set.add(4)
+sorted_set.add(6)
+print(sorted_set)  # SortedSet([1, 2, 3, 4, 5, 6])
+
+# Remove elements
+sorted_set.discard(3)
+sorted_set.discard(5)
+print(sorted_set)  # SortedSet([1, 2, 4, 6])
+
+# Check if an element is in the sorted set
+print(2 in sorted_set)  # True
+print(100 in sorted_set)  # False
+
+# Access elements by it's index
+print(sorted_set[0])  # 1 (smallest element, first index)
+print(sorted_set[-1])  # 6 (largest element, last index)
+print(sorted_set[2])  # 4
+
+# Get the index of an element
+print(sorted_set.index(4))  # 2
 
+# Find the index to insert the given value
+print(sorted_set.bisect_left(2))  # 1
+print(sorted_set.bisect_right(2))  # 2
+```
+
+<Warning>
+
+Most online judges (including USACO) do not provide the `sortedcontainers`
+library as it is not from Python's standard library.
+
+</Warning>
+
```

Comment:
```suggestion
```
alr mentioned

Reply:
not addressed?

Reply:
Do you mean it's already addressed here? `Since some online judges include additional libraries, an implementation ...` I meant it more as to why a sorted set implementation from `sortedcontainers` is shown, and the warning suits as more of a warning.

But regarding this, I also removed the warning under sorted maps, since that was already mentioned.

Reply:
i mean you should probably include that in the initial description
"there is a 3rd party `sortecontainers` library that some ojs have"

Reply:
I've added a note/warning in the initial description that just specifies that additional libraries such as `sortedcontainers` is not supported by USACO, etc.

```mdx
Since some online judges include additional libraries, an implementation of
sorted sets from the `sortedcontainers` library (which is not included in most
online judges like USACO) is shown below. All operations shown below are
$\mathcal{O}(\log N)$ time, except for its $\mathcal{O}(N \log N)$
initialization.
```